### PR TITLE
[rush] Make git optional for rush (re)build

### PR DESCRIPTION
**Pull request by iclanton**

### CODE REVIEW

File: apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts [hunk 1]
```diff
@@ -49,13 +50,24 @@ export class PackageChangeAnalyzer {
...
 
-    // Load the package deps hash for the whole repository
-    const repoDeps: IPackageDeps = PackageChangeAnalyzer.getPackageDeps(
-      this._rushConfiguration.rushJsonFolder, [RushConstants.packageDepsFilename]
-    );
+    let repoDeps: IPackageDeps;
+    try {
+      // Load the package deps hash for the whole repository
+      repoDeps = PackageChangeAnalyzer.getPackageDeps(
+        this._rushConfiguration.rushJsonFolder + 'asdfasdfasdfasdfasdf', [RushConstants.packageDepsFilename]
```
Comment (pgonzal) [inline apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts]: >asdfasdfasdfasdfasdf [](start = 50, length = 20)

Remove the "asdfasdfasdfasdfasdf" test code

File: apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts [hunk 2]
```diff
@@ -49,13 +50,24 @@ export class PackageChangeAnalyzer {
...
+      );
+    } catch (e) {
```
Comment (pgonzal) [inline apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts]: >catch (e) { [](start = 6, length = 11)

Is there a way to do this without an unexceptional exception?  For example, can we simply test whether we are in a Git repository, and disable all the Git operations if not?

**Comment by pgonzal** [comment]
>   private getData(): Map<string, IPackageDeps> {

What's the maximum number of times that `getData()` gets called during a Rush invocation?  If it's more than one, then it seems weird for this library function to be throwing exceptions and printing stuff to the console.

---
Refers to: apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts:36 in 4916728. [](commit_id = 4916728e45fa2fc43740624c07d3edab0570539d, deletion_comment = False)

File: apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts [hunk 3]
```diff
@@ -137,4 +158,23 @@ export class PackageChangeAnalyzer {
...
     return Path.isUnder(filePath, folderPath);
   }
-}
\ No newline at end of file
+
+  /**
+   * TODO: Promote this to be part of RushConfiguration, or another global configuration object
+   * if this ends up being useful elsewhere
+   */
+  private _detectIfGitIsSupported(): boolean {
```
[the remark this answers is not in the document]
Comment (iclanton) [reply apps/rush-lib/src/cli/logic/PackageChangeAnalyzer.ts]: Good idea.

**Pull request merged**